internal/datacoord/index_service.go

```diff
@@ -258,6 +276,10 @@ func (s *Server) AlterIndex(ctx context.Context, req *indexpb.AlterIndexRequest)
 
 		// update the index params
 		for _, param := range req.GetParams() {
+			if err := ValidateIndexParams(index, param.GetKey(), param.GetValue()); err != nil {
```

Comment:
you can call  funcutil.GetAttrByKeyFromRepeatedKV  to get `index_type` to avoid  unnecessary checks.